lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -0,0 +1,423 @@
+import boto3
+from datetime import datetime
+import json
+import logging
+import os
+import psycopg2
+import requests
+import sys
+
+# Global Variables
+# Constants
+START_TYPES = ('START', 'BEGIN', 'RESTART', 'OPTIN', 'OPT-IN',)
+STOP_TYPES = ('STOP', 'OPTOUT', 'OPT-OUT',)
+HELP_TYPES = ('HELP',)
+START_TEXT = 'Message service resumed, reply "STOP" to stop receiving messages.'
+STOP_TEXT = 'Message service stopped, reply "START" to start receiving messages.'
+HELP_TEXT = 'Some help text'
+INBOUND_NUMBERS_QUERY = """SELECT number, service_id, url_endpoint, self_managed FROM inbound_numbers;"""
+# Validation set
+EXPECTED_PINPOINT_FIELDS = set({'originationNumber',
+                           'destinationNumber',
+                           'messageBody'
+                           })
+
+# Pre-defined env variables
+AWS_PINPOINT_APP_ID = ''
+AWS_REGION = ''
+DEAD_LETTER_SQS_URL = ''
+LOG_LEVEL = 'INFO'
+RETRY_SQS_URL = ''
+SQLALCHEMY_DATABASE_URI = ''
+DATABASE_URI_PATH = ''
+TIMEOUT = 3
+
+logger = logging.getLogger('TwoWaySMSv2')
+logger.setLevel(logging.getLevelName(LOG_LEVEL))
+
+# Set within lambda
+aws_pinpoint_client = None
+aws_sqs_client = None
+two_way_sms_table_dict = {}
+initialized = False
+
+
+def set_env_variables() -> None:
+    """
+    Attempt to get environmental variables. Abort execution environment on failure.
+    """
+    global AWS_PINPOINT_APP_ID, AWS_REGION, DEAD_LETTER_SQS_URL
+    global LOG_LEVEL, RETRY_SQS_URL, DATABASE_URI_PATH, TIMEOUT
+
+    try:
+        AWS_PINPOINT_APP_ID = os.getenv('AWS_PINPOINT_APP_ID')
+        AWS_REGION = 'us-gov-west-1'
+        DEAD_LETTER_SQS_URL = os.getenv('DEAD_LETTER_SQS_URL')
+        LOG_LEVEL = os.getenv('LOG_LEVEL')
+        RETRY_SQS_URL = os.getenv('RETRY_SQS_URL')
+
+        TIMEOUT = os.getenv('TIMEOUT')
+
+        if isinstance(TIMEOUT, list):
+            TIMEOUT = tuple(TIMEOUT)
+        else:
+            TIMEOUT = int(TIMEOUT)
+
+        DATABASE_URI_PATH = os.getenv('DATABASE_URI_PATH')
+
+        if DATABASE_URI_PATH is None:
+            # Without this value, this code cannot know the path to the required
+            # SSM Parameter Store resource.
+            sys.exit("DATABASE_URI_PATH is not set.  Check the Lambda console.")
+    except KeyError as e:
+        sys.exit(e)
+    except Exception as e:
+        sys.exit(e)
+
+def set_service_two_way_sms_table() -> None:
+    """
+    Sets the two_way_sms_table_dict if it is not set by opening a connection to the DB and 
+    querying the table. This table should be small (a few dozen records at most).
+    """
+    # format for dict should be: {'number':{'service_id': <value>, 'url_endpoint': <value>, 'self_managed': <value> }}
+    global two_way_sms_table_dict
+
+    logger.info("Beginning retrieval of 10DLC to URL mapping")
+    
```

Comment:
```suggestion
    db_connection = None
    
```
If psycopg2 throws an exception when connecting, db_connection will not be set, so you will see a name error on line 115 when it attempts to see if it's set. An exception during the exception call. The result is the same (we exit the lambda) but it's not graceful. This line will avoid the name error.

Reply:
I refactored the code.